File: C++/28_FindTheIndexFirstOcurrenceString/Solution.cpp
#include<iostream>
#include<cstring>

int strStr(std::string haystack, std::string needle) {
  if(haystack.empty() && needle.empty()) return -1;
  return haystack.find(needle);
}

int main (void) {
  std::string haystack = "";
  std::string needle = "";

  std::cout<<strStr(haystack, needle)<<std::endl;

  return 0;
}

File: Java/28_FindTheIndexFirstOcurrenceString/Solution.java
public class Solution {
  public int strStr (String haystack, String needle) {
    if(haystack.isEmpty() && needle.isEmpty()) return -1;
    return haystack.indexOf(needle);
  }

  public static void main (String [] args) {
    String haystack="sadbutsad", needle="sad";
    Solution solution = new Solution();
    int position = solution.strStr(haystack, needle);
    System.out.println(position);
  }
}